editor/src/app/sites/sections/sections-state/site-sections.state.ts

```diff
@@ -51,31 +55,43 @@ export class SiteSectionsState implements NgxsOnInit {
 
   @Action(CreateSectionAction)
   createSection({ getState, setState }: StateContext<SiteSectionStateModel[]>, action: CreateSectionAction) {
-    const state = getState();
-    const site = this.store.selectSnapshot(AppState.getSite);
+    const siteName = this.store.selectSnapshot(AppState.getSite);
+    const data = {
+      name: action.section ? action.section.name : null,
+      site: siteName,
+      title: action.section ? action.section.title : null,
+    };
 
-    // @todo sync with backend and return section data
-    if (action.section) {
-      // clone section, pass section to backend for cloning
-    }
+    this.appStateService.sync('siteSections', data, 'POST')
+      .then((response: any) => {
```

Comment:
Let's keep to observables since we're already using them everywhere. So return observable from `sync` (remove the `toPromise`). And add `.pipe(map(response => {...}))` here.